pkg/evpn/port.go

```diff
@@ -27,117 +25,114 @@ import (
+			fmt.Printf("Failed to set MAC on link: %v", err)
+			return nil, err
+		}
 	}
-	// up
-	if err := netlink.LinkAdd(vlandev); err != nil {
-		fmt.Printf("Failed to create link: %v", err)
+	// Example: ip link set eth2 master br-tenant up
+	if err := netlink.LinkSetMaster(iface, bridge); err != nil {
+		fmt.Printf("Failed to add iface to bridge: %v", err)
 		return nil, err
 	}
-	// TODO: gap... instead of VPC we are looking for Subnet/Bridge here...
-	ifaceobj := in.Interface.Spec.GetL3IfSpec()
-	if ifaceobj.VpcNameRef != "" {
-		// Validate that a Subnet/Bridge resource name conforms to the restrictions outlined in AIP-122.
-		if err := resourcename.Validate(ifaceobj.VpcNameRef); err != nil {
-			log.Printf("error: %v", err)
+	switch in.BridgePort.Spec.Ptype {
+	case pb.BridgePortType_ACCESS:
+		// TODO: fixme
+		fmt.Println("ACCESS")
+	case pb.BridgePortType_TRUNK:
+		// Example: bridge vlan add dev eth2 vid 20 pvid untagged
+		if err := netlink.BridgeVlanAdd(iface, 66, true, true, false, false); err != nil {
+			fmt.Printf("Failed to add vlan to bridge: %v", err)
 			return nil, err
 		}
-		// now get Subnet/Bridge to plug this port/interface into
-		bridge, ok := s.Bridges[ifaceobj.VpcNameRef]
+	default:
+		msg := fmt.Sprintf("Only ACCESS or TRUNK supported and not (%d)", in.BridgePort.Spec.Ptype)
+		log.Print(msg)
+		return nil, status.Errorf(codes.InvalidArgument, msg)
+	}
+	// add port to specified logical bridges
+	for _, bridgeRefName := range in.BridgePort.Spec.LogicalBridges {
+		fmt.Printf("add iface to logical bridge %s", bridgeRefName)
+		// get object from DB
+		bridgeObject, ok := s.Bridges[bridgeRefName]
 		if !ok {
-			err := status.Errorf(codes.NotFound, "unable to find key %s", ifaceobj.VpcNameRef)
+			err := status.Errorf(codes.NotFound, "unable to find key %s", bridgeRefName)
 			log.Printf("error: %v", err)
 			return nil, err
 		}
-		brdev, err := netlink.LinkByName(path.Base(bridge.Name))
+		// find netdev
+		bridgeObjectName := path.Base(bridgeObject.Name)
+		bridgeDev, err := netlink.LinkByName(bridgeObjectName)
 		if err != nil {
-			err := status.Errorf(codes.NotFound, "unable to find key %s", bridge.Name)
+			err := status.Errorf(codes.NotFound, "unable to find key %s", bridgeObjectName)
 			log.Printf("error: %v", err)
 			return nil, err
 		}
-		if err := netlink.LinkSetMaster(vlandev, brdev); err != nil {
-			fmt.Printf("Failed to add port/interface to bridge: %v", err)
-			return nil, err
-		}
-	}
-	if err := netlink.LinkSetUp(vlandev); err != nil {
-		fmt.Printf("Failed to up link: %v", err)
-		return nil, err
-	}
-	// set MAC
-	if len(ifaceobj.MacAddress) > 0 {
-		// TODO: dummy or vlandev ?
-		if err := netlink.LinkSetHardwareAddr(dummy, ifaceobj.MacAddress); err != nil {
-			fmt.Printf("Failed to set MAC on link: %v", err)
+		// Example: ip link set eth2 master br-tenant
+		if err := netlink.LinkSetMaster(iface, bridgeDev); err != nil {
+			fmt.Printf("Failed to add iface to bridge: %v", err)
```

Comment:
I think this is not right. Not sure what are you trying to do here

Reply:
see now better ?